client-participation-alpha/src/components/InviteCodeSubmissionForm.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useState } from 'react';
+import PolisNet from '../lib/net';
+
+const submitInviteCode = (inviteCode, s, setInviteCodeReceived, conversation_id) => {
+  return async () => {
+    try {
+      const response = await PolisNet.polisPost('/inviteCode', { inviteCode });
+      if (response.success) {
+        window.localStorage.setItem(`invite-code-${conversation_id}`, inviteCode);
+        // fire custom event
+        dispatchEvent(new CustomEvent('invite-code-submitted', { bubbles: true }));
+        setInviteCodeReceived(true);
+      } else {
+        alert(s.invite_code_invalid);
+      }
+    } catch (error) {
+      alert(s.invite_code_invalid);
+      // stub code, delete later
+      window.localStorage.setItem(`invite-code-${conversation_id}`, inviteCode);
+      dispatchEvent(new CustomEvent('invite-code-submitted', { bubbles: true }));
+      setInviteCodeReceived(true);
+    }
+  };
+};
+
+
+export default function InviteCodeSubmissionForm({ s, setInviteCodeReceived, conversation_id }) {
+  const [inviteCode, setInviteCode] = useState('');
+  useEffect(() => {
+    const storedInviteCode = window.localStorage.getItem(`invite-code-${conversation_id}`);
```

Comment:
not sure if this functionality is still desired, but this is what auto submits a stored invite code if one is found in localStorage